books/views.py

```diff
@@ -1,32 +1,44 @@
 from django.http import HttpResponse, HttpResponseRedirect
 from django.views.generic import TemplateView
 from django.contrib.auth import login, authenticate
+from django.contrib.auth.decorators import login_required
 from django.contrib.auth.forms import UserCreationForm
 from django.shortcuts import render, redirect
 from django.utils import timezone
+from django.utils.decorators import method_decorator
 from .models import Book
 from .forms import PostBookForm
 
 
 class HomePageView(TemplateView):
     template_name = 'home.html'
 
+class BookNewView(TemplateView):
+    form_class = PostBookForm
+    initial = {}
```

Comment:
⚠️ **This is a common python anti-pattern** 😉 

`initial` here is defined as a **class** variable, that is, it's shared in memory between the class and all its instances.

Consider this code:
```python
>>> class A(object):
...     initial = {}
...
>>> a_one = A()
>>> a_two = A()
>>> a_one.initial['foo'] = 'bar'
>>> a_two.initial
{'foo': 'bar'}
```

In general, we do not "initialise" values in python at the class level, unless they are really meant to be **class** (and not instance) variables. To solve the problem we saw above, we could re-write the above as follows:

```python
>>> class A(object):
...     def __init__(self):
...         self.initial = {}
...
>>> a_one = A()
>>> a_two = A()
>>> a_one.initial['foo'] = 'bar'
>>> a_two.initial
{}
>>>
```

or, if `initial` is just a constant default, consider having something like:

```python
>>> class A(object):
...     def get_initial(self):
...         return {}
```

this way every time you call `get_initial()` you will get a fresh copy which is impossible to accidentally mutate. This last pattern is usually "good-practice" when you deal with **mutable** data structures (most commonly `dict` and `list`).

Reply:
However, in this particular use-case (in our view) we may not need the above at all.
I think `initial` is not really needed at all (you could instantiate the form without passing any parametres with the same result).